generate/writeToJsonFile.js

```diff
@@ -0,0 +1,82 @@
+import path from 'path'
+import fs from 'fs'
+
+const JsonStringifyParameters = {
+  replacer: null,
+  space: 4,
+}
+
+
+export function writeJsonArray(arrayNewItems, config, applyFilter = false) {
+  const DEST_JSON_FILE = path.join(__dirname, config.destDirectoryFromHere, `${config.destFileName}.json`)
+  // READ FILE
+  fs.readFile(DEST_JSON_FILE, (err, data) => {
+    if (err) {
+      if (err.code === 'ENOENT') {
+        data = JSON.stringify([])
+        fs.writeFileSync(DEST_JSON_FILE, '[]\n')
+      } else {
+        process.exit(1)
+      }
+    }
+    // ALL CONTENT INTO ARRAY
+    const allFileContent = parseReadData(data)
+
+    // UPDATE MATCHING ARRAY ITEMS (ONLY) TO CONTENT
+    arrayNewItems.map(item => {
+      const itemIndex = allFileContent.findIndex(
+        fileContent => fileContent[config.testEqualityBy] === item[config.testEqualityBy]
+      )
+      const itemToCopy = applyFilter ? filterProperties(config.propertiesToFilter, item) : item
+      if (itemIndex > -1) {
+        allFileContent.splice(itemIndex, 1, itemToCopy)
+      } else {
+        allFileContent.push(itemToCopy)
+      }
+    })
+    // WRITE UPDATED CONTENT
+    writeToFile(DEST_JSON_FILE, stringify(allFileContent))
+  })
+}
+
+export function writeJsonObject(newItem, config, applyFilter = false) {
+  const DEST_JSON_FILE = path.join(__dirname, config.destDirectoryFromHere, `${config.destFileName}.json`)
+  const itemToCopy = applyFilter ? filterProperties(config.propertiesToFilter, newItem) : newItem
+  writeToFile(DEST_JSON_FILE, stringify(itemToCopy))
+}
+
+
+function parseReadData(data) {
+  let parsedData
+  try {
+    parsedData = JSON.parse(data)
+  } catch (e) {
+    // when file already exists but is empty
+    parsedData = []
+  }
+  return parsedData
+}
+
+function writeToFile(file, content) {
+  fs.writeFile(file, content, err => {
+    if (err) {
+      process.exit(1)
```

Comment:
Since I will simplify my code to rewrite all data.json (rather updating), I won't need to read file anymore so I won't have to catch errors.